cpp/ScanChain_Hgg.C

```diff
@@ -409,6 +438,66 @@ int ScanChain_Hgg(TChain *ch, double genEventSumw, TString year, TString process
+          {
+            if (genparts[igenpart].isxyh()) {
+              GenX = genparts[igenpart].mother_p4(); 
+              gen_child_xyh.push_back(genparts[igenpart].p4());
+              GenX_pt = GenX.Pt();
+              GenX_eta = GenX.Eta();
+              GenX_phi = GenX.Phi();
+              GenX_mass = GenX.M();
+            }
+            if (genparts[igenpart].isygg()) {
+              GenY = genparts[igenpart].mother_p4(); 
+              gen_child_ygg.push_back(genparts[igenpart].p4());
+              GenY_pt = GenY.Pt();
+              GenY_eta = GenY.Eta();
+              GenY_phi = GenY.Phi();
+              GenY_mass = GenY.M();
+            }
+            if (genparts[igenpart].ishbb()) {
+              GenHiggs = genparts[igenpart].mother_p4();
+              gen_child_hbb.push_back(genparts[igenpart].p4());
+              GenHiggs_pt = GenHiggs.Pt();
+              GenHiggs_eta = GenHiggs.Eta();
+              GenHiggs_phi = GenHiggs.Phi();
+              GenHiggs_mass = GenHiggs.M();
+            }
+          }
+        TLorentzVector sort_GenPart;
+//          if (gen_child_xyh[0].Pt()<gen_child_xyh[1].Pt()) {sort_GenPart = gen_child_xyh[0]; gen_child_xyh[0]= gen_child_xyh[1]; gen_child_xyh[1] = sort_GenPart;}
+//          if (gen_child_ygg[0].Pt()<gen_child_ygg[1].Pt()) {sort_GenPart = gen_child_ygg[0]; gen_child_ygg[0]= gen_child_ygg[1]; gen_child_ygg[1] = sort_GenPart;}
+        if (!abs(GenX_pt+999)<0.0001){
+          GenX_dR = gen_child_xyh[0].DeltaR(gen_child_xyh[1]);
+        }
+        if (!abs(GenY_pt+999)<0.0001){
+          GenY_dR = gen_child_ygg[0].DeltaR(gen_child_ygg[1]);
+        }
+        if (!abs(GenHiggs_pt+999)<0.0001){
+          GenHiggs_dR = gen_child_hbb[0].DeltaR(gen_child_hbb[1]);
+          if (gen_child_hbb[0].Pt()<gen_child_hbb[1].Pt()) {sort_GenPart = gen_child_hbb[0]; gen_child_hbb[0]= gen_child_hbb[1]; gen_child_hbb[1] = sort_GenPart;}
+          GenBFromHiggs_1_pt = gen_child_hbb[0].Pt();
+          GenBFromHiggs_1_eta = gen_child_hbb[0].Eta();
+          GenBFromHiggs_1_phi = gen_child_hbb[0].Phi();
+          GenBFromHiggs_1_mass = gen_child_hbb[0].M();
+          GenBFromHiggs_2_pt = gen_child_hbb[1].Pt();
+          GenBFromHiggs_2_eta = gen_child_hbb[1].Eta();
+          GenBFromHiggs_2_phi = gen_child_hbb[1].Phi();
+          GenBFromHiggs_2_mass = gen_child_hbb[1].M();
+          if (selectedDiJet.leadJet.p4().DeltaR(gen_child_hbb[0])<=0.4 || selectedDiJet.leadJet.p4().DeltaR(gen_child_hbb[1])<=0.4) {dijet_lead_gen_match=true; n_gen_matched_in_dijet++;}
+          if (selectedDiJet.subleadJet.p4().DeltaR(gen_child_hbb[0])<=0.4 || selectedDiJet.subleadJet.p4().DeltaR(gen_child_hbb[1])<=0.4) {dijet_sublead_gen_match=true; n_gen_matched_in_dijet++;}
+          for (int ijet=0; ijet<jets.size(); ijet++)
+            if (jets[ijet].p4().DeltaR(gen_child_hbb[0])<=0.4 || jets[ijet].p4().DeltaR(gen_child_hbb[1])<=0.4)
+              n_gen_matched_jets++;
```

Comment:
In principle, this could match two reco jets to the same gen jet. Is this the way we want to make it work? Is this how it works in HiggsDNA?

Reply:
Yes, it won't avoid the situation that if both reco jets are matched to the same gen_jet. The HiggsDNA works the same way [here](https://gitlab.cern.ch/smay/HiggsDNA/-/blob/xyh_ggbb/higgs_dna/taggers/xyh_ggbb_tagger.py#L169). Maybe we can limit it that they cannot match to the same one?

Reply:
It seems in rare cases we can have reco jets matched to the same gen jet. I did a rough test with the HHbbgg samples, and in the output rootfiles scanning the GenHiggs_dR and dijet_dR. requiring both reco jets are gen_matched. command like this.
tout->Scan("GenHiggs_dR:dijet_dR", "n_gen_matched_in_dijet==2 && dijet_dR<0.8")
The two dR numbers should be roughly the same if reco dijets are matched two gen jets separately.
but I saw some events, with GenHiggs_dR=1.4574552, dijet_dR=0.5415368, seems like the dijet pairs aren't matched to two gen jets from Higgs. Another event with GenHiggs_dR=2.923605, dijet_dR=0.6584001 like this